### PR TITLE
refactoring(logitech_receiver/notifications): change to intenums some NamedInts Errors

### PR DESCRIPTION
**Pull request by rloutrel**
Replace some historic NamedInts enum for standard IntEnum

### CODE REVIEW

**Comment by rloutrel** [comment]
Hope it is OK like that. I added some tests as well. I will continue with the NamedInts from hidpp10_constants

**Comment by MattHag** [comment]
Looks good, only the additional pre-commit would clutter the commit history, but I think it can be squashed when it's merged.

Install hooks via `pre-commit install` and you shouldn't even be able to create ill formatted commits anymore.

**Review by MattHag — APPROVED** [review]

File: tests/logitech_receiver/test_base.py [hunk 1]
```diff
@@ -111,3 +118,35 @@ def test_get_next_sw_id():
...
+@pytest.mark.parametrize(
+    "simulated_error, expected_result",
+    [
+        (ERROR.invalid_SubID__command, 1.0),
+        (ERROR.resource_error, None),
+        (ERROR.connection_request_failed, None),
+        (ERROR.unknown_device, exceptions.NoSuchDevice),
+    ],
+)
+def test_ping(simulated_error, expected_result):
```
Comment (MattHag) [inline tests/logitech_receiver/test_base.py:132]: Add this to skip test on macOS or patch the hidapi to make it work on macOS
`
import pytest
import sys

@pytest.mark.skipif(sys.platform == "darwin", reason="Test only runs on Linux")
`

**Comment by rloutrel** [comment]
Sorry . It was not meant to be in this PR

**Comment by MattHag** [comment]
@pfps Let‘s get it squashed and merged

**Pull request merged**